src/Groups/matrices/linear_centralizer.jl

```diff
@@ -0,0 +1,426 @@
+# TODO: in this file are used many methods TEMPORARILY defined in files matrix_manipulation.jl and stuff_field_gen.jl
+# once methods in those files will be deleted / replaced / modified, this file need to be modified too
+
+
+########################################################################
+#
+# Orders
+#
+########################################################################
+
+function _GL_order(n::Int, q::fmpz)
+   res = q^div(n*(n-1),2)
+   for i in 1:n res *= (q^i-1) end
+   return res
+end
+
+_GL_order(n::Int, F::Ring) = _GL_order(n, order(F))
+
+function _SL_order(n::Int, q::fmpz)
+   res = q^div(n*(n-1),2)
+   for i in 2:n res *= (q^i-1) end
+   return res
+end
+
+_SL_order(n::Int, F::Ring) = _SL_order(n, order(F))
+
+########################################################################
+#
+# Centralizer in GL
+#
+########################################################################
+
+# Compute a generating set for GL(n,q) and SL(n,q)
+# generators for GL(n,q) and SL(n,q) are described in: 
+# Taylor, D. E., Pairs of Generators for Matrix Groups. I, The Cayley Bulletin, 3 (1987)
+
+# returns as matrices
+# returns a generating set for GL(n,F) of at most two elements
+function _gens_for_GL(n::Int, F::FinField)
+   n !=1 || return [matrix(F,1,1,[primitive_element(F)])]
+   h1 = identity_matrix(F,n)
+   h2 = zero_matrix(F,n,n)
+   if order(F)==2
+      h1[1,2] = 1
+   else
+      h1[1,1] = primitive_element(F)
+      h2[1,1] = -1
+   end
+   h2[1,n] = 1
+   for i in 1:n-1 h2[i+1,i] = -1 end
+   return h1,h2
```

Comment:
I believe this returns a tuple, while in case `n=1` you return a `Vector` with one element; this violates type stability. I would suggest to just always return a vector? I.e.,
```suggestion
   return [h1,h2]
```

Reply:
Done it.